### DISCUSSION TITLE
How to implement interpolation in multivalue

### DISCUSSION THREAD

**Discussion by SangXH** [Electrostatics/apbs#282] [Q&A]
Hi,
I’d like to know how multivalue implements interpolation, using the Vgrid_value function? What is the specific method of interpolation? For example, I use multivalue to interpolate the data of the vertices which represent molecular surface. How do I then estimate the value of a certain area on the surface? Is this related to the vertices' values and the density of the vertices? But I think the simple product of value and density might be wrong. So the interpolation of vertices by multivalue may affect the value of the final area. I want to know the specific function of multivalue.
Thank you!

**Comment by sobolevnrm** [comment]
Hello --

APBS uses tri-linear interpolation.  The code can be found here:

https://github.com/Electrostatics/apbs/blob/c2e5af22c9044b2bb8f9c1114710add58ec5b007/src/mg/vgrid.c#L236. 

If you have a dot representation of the surface, you should be able to call `Vgrid_value` at each surface doc.  However, you are correct that you will need to figure out how to weight the surface quadrature based on the spacing of dots on the surface.  We do not have a function that directly supports that quadrature.

Thank you,

Nathan
